cadence/server/public/index.html

```diff
@@ -0,0 +1,210 @@
+
+                <h2 class="title is-3"><span class="tag is-success block">GET</span> <code>/api/listeners</code></h2>
+                <p>Returns the number of connected listeners.</p>
+                <h3 class="title is-6">Response</h3>
+                <ul>
+                    <li>200 OK <code>{ "Listeners": 14 }</code>
+                </ul>
+
+                <hr/>
+
+                <h2 class="title is-3"><span class="tag is-success block">GET</span> <code>/api/listenurl</code></h2>
+                <p>Returns the direct audio listen URL. This is used for programmatically setting any audio sources.</p>
+                <h3 class="title is-6">Response</h3>
+                <ul>
+                    <li><code>200 OK { "ListenURL": "cadenceradio.com/cadence1" }</code></li>
+                </ul>
+
+                <hr/>
+                
+                <h2 class="title is-3"><span class="tag is-success block">GET</span> <code>/api/version</code></h2>
+                <p>Returns the server version.</p>
+                <h3 class="title is-6">Response</h3>
+                <ul>
+                    <li><code>200 OK { "Version": "5.0.0" }</code></li>
+                </ul>
+
+                <hr/>
+
+                <h2 class="title is-3"><span class="tag is-success block">GET</span> <code>/api/radiodata/sse</code></h2>
+                <p>This is a special server-sent event (SSE) connection API.</p>
+                <p>Clients may connect to this endpoint, and as long as they stay connected, will receive server-pushed data as it changes live on the backend without the need for the client to poll. The data is sent within one second after a change is observed.</p>
+                <p>All data transfer happens through only one event source and is differentiated by unique <code>event</code>s. At the moment, this event source will monitor and notify changes for <code>title</code>, <code>artist</code>, <code>listenurl</code>, and <code>listeners</code>. All of these can also be fetched on-demand through their respective API endpoints.</p>
+                <p>As an example, if the playing song changes, all connected clients receive a <code>"title"</code> and <code>"artist"</code> event from the server. Here is an example of subscribing to the event stream in JavaScript:</p>
+<pre>
+let eventSource = new EventSource("/api/radiodata/sse");
+eventSource.addEventListener("title", function(event) {
+    console.log(event.data);    // "only my railgun"
+});
+eventSource.addEventListener("artist", function(event) {
+    console.log(event.data);    // "fripSide"
+});
+eventSource.addEventListener("listenurl", function(event) {
+    console.log(event.data);    // "cadenceradio.com/cadence1"
+});
+eventSource.addEventListener("listeners", function(event) {
+    console.log(event.data);    // 14
+});
+</pre>
+            </section>
+        </div>
+            <footer>
+            <figure class="image">
+                <img src="/static/cadence.png">
```

Comment:
This, instead of loading Rock Salt and letting the browser handle it at the right resolution?

Reply:
Right. Despite any disadvantages there might be, I figured if this was meant to be a logo, it should be something custom and not based on a font. I drew the picture myself.